tests/sundials/arkode_04.cc

```diff
@@ -94,26 +95,21 @@ main(int argc, char **argv)
     return 0;
   };
 
-  ode.solve_jacobian_system = [&](const double t,
-                                  const double gamma,
-                                  const VectorType &,
-                                  const VectorType &,
-                                  const VectorType &src,
-                                  VectorType &      dst) -> int {
-    J       = 0;
-    J(0, 0) = 1;
-    J(1, 1) = 1;
-    J(2, 2) = 1 + gamma / eps;
-    J.gauss_jordan();
+  ode.jacobian_times_vector = [&](const VectorType &src,
+                                  VectorType &      dst,
+                                  double            t,
+                                  const VectorType & /*y*/,
+                                  const VectorType & /*fy*/) -> int {
```

Comment:
On second thought, the documentation of this test says that it specifically checks `solve_jacobian_system`, but you are replacing this here with `jacobian_times_vector`.

Reply:
Yes - `solve_jacobian_system()` was only for sundials < 4.0.0. Is it OK if I address this when I do the big doc update?

Reply:
Yes. But could you just update the comment at the top?

Reply:
It's ok if you addressed the comment bit in a later PR.

Reply:
I need to rebase anyway so I fixed it :)